app/services/users_service.rb

```diff
@@ -3,9 +3,11 @@
 class UsersService < BaseService
   def login(email, password)
     result.user = User.find_by(email: email)&.authenticate(password)
-    result.token = generate_token if result.user
 
     return result.fail!(code: 'incorrect_login_or_password') unless result.user
+    return result.fail!(code: 'incorrect_login_or_password') unless result.user.memberships.first.active?
```

Comment:
Since order of the membership list is unpredictable, (ordered on id field, but ids are UUIDs) with this logic, if the memberships is revoked, user cannot log-in anymore even if he has other active memberships, I'm not sure it is the intended behaviors.

What about something like:
```suggestion
    return result.fail!(code: 'incorrect_login_or_password') unless result.user.memberships.active.any?
```

Reply:
Well, to be correct we should check for the membership of the current organisation 🤔 but I get your point

I know we don't handle multi-orga yet but indeed the any seems more correct.

Reply:
I guess it also handles the re-invite of revoked member

Reply:
Went with your solution. Code is updated